README.md

```diff
@@ -686,6 +686,7 @@ The OpenTelemetry Operator *might* work on versions outside of the given range,
 
 | OpenTelemetry Operator | Kubernetes           | Cert-Manager        |
 |------------------------|----------------------|---------------------|
+| v0.91.0                | v1.23 to v1.29       | v1                  |
```

Comment:
By convention, we delete the last row of this table as well, though I don't recall if this is documented.